site/src/pages/TemplatePage/TemplateInsightsPage/TemplateInsightsPage.tsx

```diff
@@ -276,6 +288,66 @@ const UserLatencyPanel = ({
   );
 };
 
+const UsersActivityPanel = ({
+  data,
+  ...panelProps
+}: PanelProps & { data: UserActivityInsightsResponse | undefined }) => {
+  const users = data?.report.users;
+
+  return (
+    <Panel {...panelProps} sx={{ overflowY: "auto", ...panelProps.sx }}>
+      <PanelHeader>
+        <PanelTitle sx={{ display: "flex", alignItems: "center", gap: 1 }}>
+          Activity by user
+          <HelpTooltip size="small">
+            <HelpTooltipTitle>How is activity calculated?</HelpTooltipTitle>
+            <HelpTooltipText>
+              When a connection is initiated to a user&apos;s workspace they are
+              considered an active user. e.g. apps, web terminal, SSH
+            </HelpTooltipText>
+          </HelpTooltip>
+        </PanelTitle>
+      </PanelHeader>
+      <PanelContent>
+        {!data && <Loader sx={{ height: "100%" }} />}
```

Comment:
I don't like checking just `data` for loading. we do this in a lot of places, and it ends up with lots of places that show spinners instead of error messages.

Reply:
I usually handle errors when they are expected like form errors, or possible user actions that can cause an "invalid" state. IMO handling all the errors caused by async can be too much and idk if that adds too much value, to be honest, but I'm open to that.

Maybe could be interesting to start a thread on dev about handling errors in the UI and see if we can come up with a more solid strategy.

Reply:
I'm also going to put some thoughts on that 🤔. Can we get this work merged and I open the PR to handle the error for all the Insights pages?

Reply:
fair enough

Reply:
Is there anything else I should do to get your quality approval? 👀

Reply:
Not going to treat this as a blocking comment, but does the whole data need to be made a prop, or would it be safe just to have the component receive the users, since that's the only part the component seems like it would ever care about?

Reply:
Yes, it could but since its goal is to display the `UserActivityInsightsResponse` data I think it is ok to use it. 

I think it is just easier to maintain and extend using the context object than passing prop by prop.